neurons/miner3.py

```diff
@@ -65,16 +74,23 @@ async def evaluate_model(self) -> None:
         if self.config.clean_after_run:
             dataset_manager.delete_dataset()
 
+    async def compress_code(self) -> str:
+        bt.logging.info("Compressing code")
+        out, err = await run_command(
+            f"zip  {self.config.code_directory}/code.zip {self.config.code_directory}/*"
+        )
+        return f"{self.config.code_directory}/code.zip"
+
     async def submit_model(self) -> None:
-         # The wallet holds the cryptographic key pairs for the miner.
-                bt.logging.info(
-                    f"Initializing connection with Bittensor subnet {self.config.netuid} - Safe-Scan Project"
-                )
-                bt.logging.info(f"Subtensor network: {self.config.subtensor.network}")
-                bt.logging.info(f"Wallet hotkey: {self.config.wallet.hotkey.ss58_address}")
-                wallet = bt.wallet(config=self.config)
-                subtensor = bt.subtensor(config=self.config)
-                metagraph = subtensor.metagraph(self.config.netuid)
+        # The wallet holds the cryptographic key pairs for the miner.
+        bt.logging.info(
+            f"Initializing connection with Bittensor subnet {self.config.netuid} - Safe-Scan Project"
+        )
+        bt.logging.info(f"Subtensor network: {self.config.subtensor.network}")
+        bt.logging.info(f"Wallet hotkey: {self.config.wallet.hotkey.ss58_address}")
+        wallet = bt.wallet(config=self.config)
+        subtensor = bt.subtensor(config=self.config)
+        metagraph = subtensor.metagraph(self.config.netuid)
```

Comment:
shouldnt be like this?
```suggestion
        wallet = self.wallet
        subtensor = self.subtensor
        metagraph = self.metagraph
```
I recon it in the base neuron class, but I guess I will integrate so no need to change it now, thought it would be nice to leave a TODO comment to make life easier for future me